tt_metal/hw/inc/ethernet/dataflow_api.h

```diff
@@ -232,9 +283,53 @@ void eth_send_bytes(
             0, ((num_bytes_sent + src_addr) >> 4), ((num_bytes_sent + dst_addr) >> 4), num_bytes_per_send_word_size);
         num_bytes_sent += num_bytes_per_send;
     }
-    erisc_info->user_buffer_bytes_sent += num_bytes;
+    erisc_info->per_channel_user_bytes_send[0].bytes_sent += num_bytes;
 }
 
+/**
+ * Initiates an asynchronous write from a source address in L1 memory on the local ethernet core to L1 of the connected
+ * remote ethernet core. Also, see \a eth_is_receiver_channel_send_done and \a eth_bytes_are_available_on_channel.
+ *
+ * Non-blocking
+ *
+ * Return value: None
+ *
+ * | Argument                    | Description                                             | Type     | Valid Range | Required |
+ * |-----------------------------|---------------------------------------------------------|----------|-------------|----------|
+ * | src_addr                    | Source address in local eth core L1 memory              | uint32_t | 0..256kB    | True     |
+ * | dst_addr                    | Destination address in remote eth core L1 memory        | uint32_t | 0..256kB    | True     |
+ * | num_bytes                   | Size of data transfer in bytes, must be multiple of 16  | uint32_t | 0..256kB    | True     |
+ * | channel                     | Which transaction channel to use. Corresponds to        | uint32_t | 0..7        | True     |
+ * |                             | per_channel_user_bytes_send in erisc_info_t             |          |             |          |
+ * | num_bytes_per_send          | Number of bytes to send per packet                      | uint32_t | 16..1MB     | False    |
+*  | num_bytes_per_send_word_size| num_bytes_per_send shifted right 4                      | uint32_t | 1..256kB    | False    |
+ */
+FORCE_INLINE
+void eth_send_bytes_over_channel(
+    uint32_t src_addr,
+    uint32_t dst_addr,
+    uint32_t num_bytes,
+    uint32_t channel,
+    uint32_t num_bytes_per_send = 16,
+    uint32_t num_bytes_per_send_word_size = 1) {
+    // assert(channel < 4);
+    uint32_t num_bytes_sent = 0;
+    while (num_bytes_sent < num_bytes) {
+        internal_::eth_send_packet(
+            0, ((num_bytes_sent + src_addr) >> 4), ((num_bytes_sent + dst_addr) >> 4), num_bytes_per_send_word_size);
+        num_bytes_sent += num_bytes_per_send;
```

Comment:
I think since you're managing multiple outstanding eth_send_bytes with channels, do we need `num_bytes_sent` anymore? Can it just be `data_sent` as 0 or 1?

Reply:
So I know we originally talked and were thinking we should just convert this to a bool. I kept it for back-compat. 

I think it's still also useful for other use cases. Just because my erisc datamover uses it as a binary status, doesn't mean we wouldn't want to allow a use-case where we jus t read the # bytes sent to enable more precise sends (say if erisc does more than just signal to workers of data available).

Especially for low latency, I imagine this would be really useful (if sender needs/wants to break into sub-packets, they can do so and receiver can start sending *right away*).

Overall, not confident enough to convert this to bool.

Reply:
Right, leave as is then